flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/cache/FileBasedCache.java

```diff
@@ -46,12 +52,37 @@ public class FileBasedCache extends LruCache<String, FileCacheEntry> {
     /** Whether the cache is closed. */
     private volatile boolean closed;
 
+    private MetricGroup metricGroup;
+
+    /** Hit metric. */
+    private transient Counter hitCounter;
+
+    /** Miss metric. */
+    private transient Counter missCounter;
+
+    /** Cached bytes metric. */
+    private transient Gauge<Long> cachedBytesGauge;
+
     public FileBasedCache(
-            int capacity, CacheLimitPolicy cacheLimitPolicy, FileSystem cacheFs, Path basePath) {
+            int capacity,
+            CacheLimitPolicy cacheLimitPolicy,
+            FileSystem cacheFs,
+            Path basePath,
+            MetricGroup metricGroup) {
         super(capacity, cacheLimitPolicy);
         this.closed = false;
         this.cacheFs = cacheFs;
         this.basePath = basePath;
+        if (metricGroup != null) {
+            this.metricGroup = metricGroup;
+            this.hitCounter =
+                    metricGroup.counter(FORST_CACHE + ".hit", new ThreadSafeSimpleCounter());
+            this.missCounter =
+                    metricGroup.counter(FORST_CACHE + ".miss", new ThreadSafeSimpleCounter());
+            this.cachedBytesGauge =
+                    metricGroup.gauge(
+                            FORST_CACHE + ".cachedBytes", () -> cacheLimitPolicy.cachedBytes());
```

Comment:
Besides this, each policy may have its' own metrics. e.g. The `SpaceBasedCacheLimitPolicy` may need report the remaining space.